test/libsolidity/SemanticTest.cpp

```diff
@@ -340,13 +379,31 @@ void SemanticTest::printUpdatedSettings(ostream& _stream, string const& _linePre
 
 void SemanticTest::parseExpectations(istream& _stream)
 {
-	TestFileParser parser{_stream};
+	TestFileParser parser{_stream, &m_builtins};
 	auto functionCalls = parser.parseFunctionCalls(m_lineOffset);
-	std::move(functionCalls.begin(), functionCalls.end(), back_inserter(m_tests));
+	move(functionCalls.begin(), functionCalls.end(), back_inserter(m_tests));
 }
 
-bool SemanticTest::deploy(string const& _contractName, u256 const& _value, bytes const& _arguments, map<string, solidity::test::Address> const& _libraries)
+bool SemanticTest::deploy(
+	string const& _contractName,
+	u256 const& _value,
+	bytes const& _arguments,
+	map<string, solidity::test::Address> const& _libraries
+)
 {
 	auto output = compileAndRunWithoutCheck(m_sources.sources, _value, _contractName, _arguments, _libraries);
 	return !output.empty() && m_transactionSuccessful;
 }
+
+std::optional<bytes> SemanticTest::builtinSmokeTest(FunctionCall const& call)
```

Comment:
Why is this not inside SmokeBuiltin?

Reply:
Oh that's the actual function! Maybe it's enough to make this a lambda function inside the constructor? Or maybe don't have it at all and 'test' the feature with an actual builtin?

Reply:
I was also thinking about this. I moved that now into a lambda, maybe we can remove it later, when we have a real builtin.